### PR TITLE
LimitOrder CancelAfter option added

### PR DESCRIPTION
**Pull request by njannink**

### CODE REVIEW

File: src/Lykke.Service.HFT.Core/Validation/CancelAfterValidationRule.cs [hunk 1]
```diff
@@ -0,0 +1,10 @@
+﻿using System;
+
+namespace Lykke.Service.HFT.Core.Validation
+{
+    public static class CancelAfterValidationRule
+    {
+        public static bool IsValid(DateTime dateTime)
+            => (dateTime - DateTime.Now).TotalSeconds > double.Epsilon;
```
Comment (polonzer) [inline src/Lykke.Service.HFT.Core/Validation/CancelAfterValidationRule.cs]: >DateTime.Now
Please always use DateTime.UtcNow.
Also I suppose the more convenient way is to compare to TimeSpan, like
dateTime - DateTime.UtcNow > TimeSpan.Zero

File: src/Lykke.Service.HFT/Controllers/OrdersController.cs [hunk 2]
```diff
@@ -68,7 +68,7 @@ public class OrdersController : Controller
 
             var clientId = User.GetUserId();
 
-            var orders = _orderStateCache.All().Where(x => x.ClientId == clientId); // todo: fix default limit of 101 element
+            var orders = await _orderStateCache.FilterAsync(x => x.ClientId == clientId); // todo: fix default limit of 101 element
```
Comment (polonzer) [inline src/Lykke.Service.HFT/Controllers/OrdersController.cs]: This change is not related to the task. Moreover it leads to client-side filtering. FilterAsync returns IEnumerable, not IQueryable.

File: src/Lykke.Service.HFT/Middleware/DoubleGreaterThanZeroAttribute.cs [hunk 3]
```diff
@@ -1,28 +1,20 @@
...
-            object value, ValidationContext validationContext)
-        {
-            if (IsValid(value))
-                return ValidationResult.Success;
-            else
-                return new ValidationResult(
-                    FormatErrorMessage(validationContext.MemberName)
-                );
-        }
+            => !(value is double price) || PriceValidationRule.IsValid(price);
```
Comment (polonzer) [inline src/Lykke.Service.HFT/Middleware/DoubleGreaterThanZeroAttribute.cs]: Absolutely not critical, but this attribute is about checking any (double) properties, not just 'price'.

File: src/Lykke.Service.HFT/Controllers/OrdersController.cs (hunk 2 above)
Comment (njannink) [reply src/Lykke.Service.HFT/Controllers/OrdersController.cs]: ok will remove

File: src/Lykke.Service.HFT.Core/Validation/CancelAfterValidationRule.cs (hunk 1 above)
Comment (njannink) [reply src/Lykke.Service.HFT.Core/Validation/CancelAfterValidationRule.cs]: fixed

File: src/Lykke.Service.HFT/Middleware/DoubleGreaterThanZeroAttribute.cs (hunk 3 above)
Comment (njannink) [reply src/Lykke.Service.HFT/Middleware/DoubleGreaterThanZeroAttribute.cs]: renamed
